src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +565,69 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        static HashSet<string> GetValidArguments(CommandLineApplication app)
+        {
+            var validArguments = new HashSet<string>();
+            foreach (var option in app.GetOptions())
+            {
+                if (!string.IsNullOrEmpty(option.LongName))
+                {
+                    validArguments.Add(option.LongName);
+                }
+                if (!string.IsNullOrEmpty(option.ShortName))
+                {
+                    validArguments.Add(option.ShortName);
+                }
+            }
+
+            return validArguments;
+        }
+
+        static ReadOnlyMemory<char> GetArgumentName(string arg) => arg.StartsWith("--") ? arg.AsMemory(2) : arg.StartsWith('-') ? arg.AsMemory(1) : ReadOnlyMemory<char>.Empty;
+        static IEnumerable<ReadOnlyMemory<char>> GetArgumentNames(IEnumerable<string> args)
+        {
+            bool lastWasArgument = false;
+            foreach (ReadOnlyMemory<char> potentialArgument in args.Select(GetArgumentName))
+            {
+                if (!lastWasArgument)
+                {
+                    bool isCurrentArgument = lastWasArgument = !potentialArgument.IsEmpty;
+                    if (isCurrentArgument)
+                    {
+                        yield return potentialArgument;
+                    }
+                }
+                else
+                {
+                    lastWasArgument = false;
+                }
+            }
+        }
+
+        // Get all valid options from the configuration files
+        var validArguments = GetValidArguments(app);
+
+        var argumentsNamesProvided = GetArgumentNames(args).ToList();
```

Comment:
I would either:
1. Leave it as `IEnumerable` and drop `ToList` - it can complain about re-enumeerating it, but as it is all in-memory there is no big downside
2. As it is local variable, use pooled memory - to avoid creating temp object:
```suggestion
        using ArrayPoolList<ReadOnlyMemory<char>> argumentsNamesProvided = GetArgumentNames(args).ToPooledList();
```

Reply:
When I drop the `ToList` and the reason why I'm using it is to avoid this error
<img width="570" alt="image" src="https://github.com/user-attachments/assets/10a2dd82-aaad-4b3e-ba53-30d00f857892">

Reply:
Using option 2 `using ArrayPoolList<ReadOnlyMemory<char>> argumentsNamesProvided = GetArgumentNames(args).ToPooledList();` gives me this other error
<img width="570" alt="image" src="https://github.com/user-attachments/assets/43d5c8fd-7507-4beb-b192-01a771ec642d">

Reply:
I decided to go with option 1 and replaced that line with the following
```
IEnumerable<ReadOnlyMemory<char>> argumentsNamesProvided = GetArgumentNames(args);
```

Reply:
`ToPooledList` is an extension method in `Nethermind.Core.Extensions.EnumerableExtensions`